tsl/src/nodes/decompress_chunk/decompress_chunk.c

```diff
@@ -359,6 +374,187 @@ cost_decompress_chunk(Path *path, Path *compressed_path)
+ * are decompressed, sorted, and then the top elements are taken from the result.
+ *
+ * The idea is to do something similar to the MergeAppend node; a BinaryHeap is used
+ * to merge the per segment by column sorted individual batches into a sorted result. So, we end
+ * up which a data flow which looks as follows:
+ *
+ * DecompressChunk
+ *   * Decompress Batch 1
+ *   * Decompress Batch 2
+ *   * Decompress Batch 3
+ *       [....]
+ *   * Decompress Batch N
+ *
+ * Using the presorted batches, we are able to open these batches dynamically. If we don't presort
+ * them, we would have to open all batches at the same time. This would be similar to the work the
+ * MergeAppend does, but this is not needed in our case and we could reduce the size of the heap and
+ * the amount of parallel open batches.
+ *
+ * The algorithm works as follows:
+ *
+ *   (1) A sort node is placed below the decompress scan node and on top of the scan
+ *       on the compressed chunk. This sort node uses the min/max values of the 'order by'
+ *       columns from the metadata of the batch to get them into an order which can be
+ *       used to merge them.
+ *
+ *       [Scan on compressed chunk] -> [Sort on min/max values] -> [Decompress and merge]
+ *
+ *       For example, the batches are sorted on the min value of the 'order by' metadata
+ *       column: [0, 3] [0, 5] [3, 7] [6, 10]
+ *
+ *   (2) The decompress chunk node initializes a binary heap, opens the first batch and
+ *       decompresses the first tuple from the batch. The tuple is put on the heap. In addition
+ *       the opened batch is marked as the most recent batch (MRB).
+ *
+ *   (3) As soon as a tuple is requested from the heap, the following steps are performed:
+ *       (3a) If the heap is empty, we are done.
+ *       (3b) The top tuple from the heap is taken. It is checked if this tuple is from the
+ *            MRB. If this is the case, the next batch is opened, the first tuple is decompressed,
+ *            placed on the heap and this batch is marked as MRB. This is repeated until the
+ *            top tuple from the heap is not from the MRB. After the top tuple is not from the
+ *            MRB, all batches (and one ahead) which might contain the most recent tuple are
+ *            opened and placed on the heap.
+ *
+ *            In the example above, the first three batches are opened because the first two
+ *            batches might contain tuples with a value of 0.
+ *       (3c) The top element from the heap is removed, the next tuple from the batch is
+ *            decompressed (if present) and placed on the heap.
+ *       (3d) The former top tuple of the heap is returned.
+ *
+ * This function checks if the compression 'order by' and the query 'order by' are
+ * compatible and the optimization can be used.
+ */
+static MergeBatchResult
+is_able_to_use_batch_merge_append(PlannerInfo *root, CompressionInfo *info, Chunk *chunk)
```

Comment:
```suggestion
can_batch_merge_append(PlannerInfo *root, CompressionInfo *info, Chunk *chunk)
```
We dont have any function that has is_able_to_use as name prefix in our code base and personally I prefer more concise function names.

Reply:
As suggested, I renamed the function to `can_batch_merge_append`